Format markdown & add better display for examples

juraj-hrivnak opened the pull request:
- Formated markdown.
- Replaced all headers in examples with `!!! example`.
- Added better example for Minecraft recipe builders.
- Increased font size in example blocks to be the same as regular code blocks.
- Added option to autohide the main header when you scroll down.

WaitingIdly commented on mkdocs.yml:
```diff
@@ -260,6 +260,7 @@ theme:
         name: Switch to dark mode
 
   features:
+    - header.autohide
```
when the header autohides, the navigation bar doesnt move up with it, leaving a rather large area weirdly empty. is this possible to fix?

juraj-hrivnak replied:
I've removed this option for now. I will make a separate PR when I figure it out.

WaitingIdly commented on docs/groovyscript/mods/thaumcraft/Infusion.md:
````diff
@@ -55,7 +55,7 @@ Register recipe: (returns nothing)
-### Example
+!!! example
 
 ```groovy
````
needs to be indented, doesnt display properly currently

WaitingIdly reviewed (commented):
a few of examples werent converted due to not being a header or being non-standard text. They include a few things in the vanilla minecraft folder, including `ore_dictionary.md`, `tooltip.md`, and `aa_events.md`, along with most things in the gregtech folder.

juraj-hrivnak commented:
Converted the remaining examples in `ore_dictionary.md`, `tooltip.md`, `aa_events.md` and those I missed.
And added a copy-to-clipboard button for code blocks.
